lib/Doctrine/Import/Builder.php

```diff
@@ -396,12 +396,11 @@ public function buildTableDefinition(array $definition)
     /**
      * buildSetUp
      *
-     * @param  array $options
-     * @param  array $columns
-     * @param  array $relations
+     * @param  array $definition
+     * @param  array $actAsColumns
      * @return string
      */
-    public function buildSetUp(array $definition)
+    public function buildSetUp(array $definition, array &$actAsColumns = array())
```

Comment:
`$actAsColumns` does it can be a private property, always initialized on `buildDefinition()`?

Reply:
Yes and no, it's already defined in buildDefinition, but I have to add it as an optional argument to any public method in the stack to avoid an BC change.

Reply:
Maybe I misunderstood you with the previous version. I had modified `array $definition` to `array &$definition` which would possibly cause a BC change, as the behaviour of the public method `buildSetUp` is changed. This is now the same public method and adding `array &$actAsColumns` as an non optional argument would also introduce a possible BC change?

Reply:
The step you have done is good, but it can be better.

It is like a storage, that must be empty each time `buildDefinition()` is called? 

Moving arguments `array &$actAsColumns` to private instance property.
This way no need anymore pass `$actAsColumns` by reference to all methods.

Reply:
Well, I'm fine with both ways. Wouldn't call it better per se. There are arguments for both ways. I've updated the pr to reflect your point.

Reply:
Yes, you got my point.

I feel that the diff and the code are more readable.

Regarding the behavior, tests still passes.

Even its coverage is low, it covers the main behavior, it is ?

Reply:
How to see the coverage? I'm only familiar with phpunit, but thats not in use here. - Anyway, not every case is covered in the test, that's correct.

Reply:
To test code coverage.

1. modify production code behaviour, (e.g. remove early return)
2. execute tests suite
3. if test suite passes, then the deleted production code was not covered

Reply:
Ah, thought you have some magic tool I can learn from :)

While I didn't really see any risky paths in it, I covered the determineActAsColumns-Method completely now.

Reply:
@thirsch  I started to investigate if it is possible to convert the test system to phpunit and found Coverage.php in the tests/DoctrineTest folder.

Reply:
Wonderful! After digging a bit in the code, I got it running and found one more case to test. If interested, I can open another pr to fix the coverage and maybe we can even add  the coverage report to our build as well.

Reply:
You make the coverage working, that's awesome.

As on master branch, the docker env is waiting for https://github.com/FriendsOfSymfony1/doctrine1/pull/90 be merged.

What test environment do you use locally?

Reply:
For my coverage testing experiments, I've just used php82 from what's already in the repo.